packages/synapse-react-client/src/components/SynapseChat/SynapseChatInteraction.tsx

```diff
@@ -54,41 +92,88 @@ export function SynapseChatInteraction({
       >
         <ListItemText primary={userMessage} />
       </ListItem>
-      {textContent && (
-        <ListItem
+      <ListItem
+        sx={{
+          display: 'grid',
+          gridTemplateColumns: '50px auto',
+          columnGap: '0px',
+          justifyItems: 'center',
+          alignItems: 'start',
+          p: '0px',
+        }}
+      >
+        <Box
           sx={{
-            display: 'grid',
-            gridTemplateColumns: '50px auto',
-            columnGap: '0px',
-            justifyItems: 'center',
-            alignItems: 'start',
-            p: '0px',
+            p: '3px',
+            mt: '10px',
+            height: '31px',
+            borderRadius: '50%',
+            borderStyle: 'solid',
+            borderWidth: '1px',
+            borderColor: 'grey.300',
           }}
         >
-          <Box
-            sx={{
-              p: '3px',
-              mt: '10px',
-              height: '31px',
-              borderRadius: '50%',
-              borderStyle: 'solid',
-              borderWidth: '1px',
-              borderColor: 'grey.300',
+          <SmartToyTwoTone sx={{ color: 'secondary.main' }} />
+        </Box>
+        <Box
+          sx={{
+            borderRadius: '10px',
+            padding: '10px',
+            maxWidth: '100%',
+          }}
+        >
+          <Button
+            variant={'outlined'}
+            size={'small'}
+            startIcon={
+              isLoading ? (
+                <SynapseSpinner size={14} />
+              ) : showTrace ? (
+                <KeyboardArrowDown sx={{ width: '14px' }} />
+              ) : (
+                <KeyboardArrowRight sx={{ width: '14px' }} />
+              )
+            }
+            disabled={!hasTraceInfo}
+            onClick={() => {
+              setShowTrace(v => !v)
             }}
-          >
-            <SmartToyTwoTone sx={{ color: 'secondary.main' }} />
-          </Box>
-          <Box
             sx={{
-              borderRadius: '10px',
-              padding: '10px',
-              maxWidth: '100%',
+              height: '20px',
+              fontSize: '12px',
+              fontWeight: 600,
+              mb: 1,
+              border: 'none !important',
+              color: 'grey.700',
+              // width: '100%',
+              justifyContent: 'flex-start',
+              whiteSpace: 'nowrap',
+              maxWidth: '325px',
+              // fontSize: '14px',
+              textTransform: 'none',
             }}
           >
-            <MarkdownSynapse markdown={textContent} />
-          </Box>
-        </ListItem>
-      )}
+            <Box sx={{ overflow: 'hidden', textOverflow: 'ellipsis' }}>
+              {traceButtonText}
+            </Box>
+          </Button>
+          {chatResponseTrace && (
+            <Collapse in={showTrace}>
+              <MarkdownSynapse
+                markdown={
+                  '<blockquote>' +
```

Comment:
I don't think the blockquote UI makes sense here, but would defer to Adam/Shysta to make a recommendation.  I'd prefer a container box with a flat background (and perhaps modifying the default trace text color/size).